### DISCUSSION TITLE
How do I get a list of default options used by casadi for IPOPT?

### DISCUSSION THREAD

**Discussion by sandeep026** [casadi/casadi#3506] [Q&A] [answered]
IPOPT has more than 280+ options for the solver. I would like to know whether the the defaults mentioned on the IPOPT website is used in casadi. Is there a way to view these deafaults via any methods?

**Comment by jgillis** [comment, marked as answer]
We let IPOPT intialize itself with its own default options, really. So the IPOPT website is correct.
The only exception is `hessian_approximation`, which we put to exact by default.

There's probably an option to print out the list of current option values, too.
